Buildings/Experimental/DHC/EnergyTransferStations/Combined/Generation5/ChillerBorefield.mo

```diff
@@ -66,6 +71,45 @@ model ChillerBorefield
     "Record with performance data for evaporator pump"
     annotation (Dialog(group="Chiller"),choicesAllMatching=true,
     Placement(transformation(extent={{100,222},{120,242}})));
+  parameter Modelica.SIunits.PressureDifference dp1WSE_nominal(
+    displayUnit="Pa") = 40E3
+    "Nominal pressure drop across heat exchanger on district side"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  parameter Modelica.SIunits.PressureDifference dp2WSE_nominal(
+    displayUnit="Pa") = 40E3
+    "Nominal pressure drop across heat exchanger on building side"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  parameter Modelica.SIunits.HeatFlowRate QWSE_flow_nominal = 0
+    "Nominal heat flow rate through heat exchanger (<=0)"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  parameter Modelica.SIunits.Temperature T_a1WSE_nominal = 6 + 273.15
+    "Nominal water inlet temperature on district side"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  parameter Modelica.SIunits.Temperature T_b1WSE_nominal = 11 + 273.15
+    "Nominal water outlet temperature on district side"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  parameter Modelica.SIunits.Temperature T_a2WSE_nominal = 15 + 273.15
+    "Nominal water inlet temperature on building side"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  parameter Modelica.SIunits.Temperature T_b2WSE_nominal = 8 + 273.15
+    "Nominal water outlet temperature on building side"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  parameter Real y1WSEMin(final unit="1") = 0.05
+    "Minimum pump flow rate or valve opening for temperature measurement (fractional)"
+    annotation (Dialog(group="Waterside economizer", enable=have_WSE));
+  replaceable parameter Buildings.Fluid.Movers.Data.Generic perPum1WSE(
+      motorCooledByFluid=false) constrainedby
+    Buildings.Fluid.Movers.Data.Generic
+    "Record with performance data for primary pump of waterside economizer"
+    annotation (
+    Dialog(group="Waterside economizer", enable=not have_val1Hex and have_WSE),
+    choicesAllMatching=true,
+    Placement(transformation(extent={{220,222},{240,242}})));
+  final Modelica.SIunits.MassFlowRate m1WSE_flow_nominal=
+    abs(QWSE_flow_nominal/4200/(T_b1WSE_nominal - T_a1WSE_nominal))
```

Comment:
The reason it is here is because when assembling a whole DHC system model, on might want to access that value to size the service and main lines. However since WSE is conditional, this parameter cannot be used outside of connect clauses par Modelica specification. 
The reason why cp is literal and not computed is because it is likely to be used to assign the nominal attribute of pressure drop variables in valves and pipes. This will trigger a warning in Dymola as only literals can be used for such assignments.
I agree that this is not ideal. A proper class interface such as the ones developed in Linkage solves most of those issues.

Reply:
Ok great, that makes sense. As long as it is justified, I don't have any problem with it. I just wanted to make sure that this wasn't an oversight.